src/coreComponents/linearAlgebra/solvers/RichardsonSolver.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file RichardsonSolver.hpp
+ */
+
+#ifndef GEOS_LINEARALGEBRA_SOLVERS_RICHARDSONSOLVER_HPP_
+#define GEOS_LINEARALGEBRA_SOLVERS_RICHARDSONSOLVER_HPP_
+
+#include "linearAlgebra/solvers/KrylovSolver.hpp"
+
+namespace geos
+{
+
+/**
+ * @brief Implements right-preconditioned modified Richardson iteration.
```

Comment:
```suggestion
 * @brief Implements preconditioned Richardson iteration.
```
The preconditioner is applied to the residual, so it is left preconditioning. Commonly this scheme is called simply preconditioned Richardson iteration.

Reply:
Updated